// @electron-forge/plugin-webpack/dist/util/rendererTypeUtils.js
"use strict";
/* eslint-disable @typescript-eslint/no-explicit-any */
Object.defineProperty(exports, "__esModule", { value: true });
exports.isPreloadOnlyEntries = exports.isLocalOrNoWindowEntries = exports.hasPreloadScript = exports.isNoWindow = exports.isPreloadOnly = exports.isLocalWindow = void 0;
/**
 * Reusable type predicate functions to narrow down the type of the WebpackPluginEntryPoint
 */
const isLocalWindow = (entry) => {
    return !!entry.html;
};
exports.isLocalWindow = isLocalWindow;
const isPreloadOnly = (entry) => {
    return !entry.html && !entry.js && !!entry.preload;
};
exports.isPreloadOnly = isPreloadOnly;
const isNoWindow = (entry) => {
    return !entry.html && !!entry.js;
};
exports.isNoWindow = isNoWindow;
const hasPreloadScript = (entry) => {
    return 'preload' in entry;
};
exports.hasPreloadScript = hasPreloadScript;
const isLocalOrNoWindowEntries = (entries) => {
    for (const entry of entries) {
        if (!(0, exports.isLocalWindow)(entry) && !(0, exports.isNoWindow)(entry)) {
            return false;
        }
    }
    return true;
};
exports.isLocalOrNoWindowEntries = isLocalOrNoWindowEntries;
const isPreloadOnlyEntries = (entries) => {
    for (const entry of entries) {
        if (!(0, exports.hasPreloadScript)(entry)) {
            return false;
        }
    }
    return true;
};
exports.isPreloadOnlyEntries = isPreloadOnlyEntries;
